src/api/entities/HistoricalMultiSigProposal.ts

```diff
@@ -0,0 +1,177 @@
+import BigNumber from 'bignumber.js';
+
+import { Context, Entity } from '~/internal';
+import { multiSigProposalVotesQuery } from '~/middleware/queries';
+import { Query } from '~/middleware/types';
+import { MultiSigProposalVote, SignerType } from '~/types';
+import { Ensured } from '~/types/utils';
+import { middlewareEventDetailsToEventIdentifier, signerValueToSigner } from '~/utils/conversion';
+
+export type MiddlewareMultiSigProposal = {
+  id: string;
+  proposalId: number;
+  multisigId: string;
+  approvalCount: number;
+  rejectionCount: number;
```

Comment:
If this is to be exported, then return type should be BigNumber instead of number